app/trip.py

```diff
@@ -0,0 +1,112 @@
+from __future__ import annotations
+import math
+from typing import List
+from importlib import import_module
+
+
+class Trip:
+
+    def __init__(
+        self,
+        shop_name: str,
+        shop_location: list,
+        trip_price: float,
+        total_milk_cost: float,
+        total_bread_cost: float,
+        total_butter_cost: float
+    ) -> None:
+        self.shop_name = shop_name
+        self.shop_location = shop_location
+        self.trip_price = trip_price
+        self.total_milk_cost = total_milk_cost
+        self.total_bread_cost = total_bread_cost
+        self.total_butter_cost = total_butter_cost
+        self.total_product_price = sum(
+            [total_milk_cost, total_bread_cost, total_butter_cost]
+        )
+        self.total_trip = self.trip_price + self.total_product_price
+
+    @staticmethod
+    def cheapest_trip(trips: List[Trip]) -> Trip:
+        return min(trips, key=lambda obj: obj.total_trip)
+
+    @staticmethod
+    def trip_distance(
+        shop_location: List[int],
+        customer_location: List[int]
+    ) -> float:
+        return math.sqrt(
+            (shop_location[0] - customer_location[0]) ** 2
+            + (shop_location[1] - customer_location[1]) ** 2
+        )
+
+    @staticmethod
+    def trip_price(
+        fuel_consumption: float,
+        fuel_price: float,
+        trip_distance: float
+    ) -> float:
+        return (
+            fuel_consumption
+            * fuel_price
+            * trip_distance / 100
+        ) * 2
+
+    @staticmethod
+    def choose_shop_and_go_to_trip(
+        customer_instances: list,
+        shop_instances: list
+    ) -> None:
+        class_customer = import_module("app.customer").Customer
+        class_shop = import_module("app.shop").Shop
```

Comment:
Why you need it?

Reply:
"I used importlib because there was an error with cyclic imports of the Customer and Shop modules.
But now I fixed it too